### PR TITLE
Control scheme refactor

### PR DESCRIPTION
**Pull request by DomNomNom**

### CODE REVIEW

File: bot_code/modelHelpers/actions/action_factory.py [hunk 1]
```diff
@@ -3,52 +3,108 @@
...
-                      []]
+'''
+ranges:
+  A range is a configuration object which holds information about its name, possible values and loss function.
+  Currently represented as a tuple:
+  - Name (string)
+  - Arguments to np.arange (tuple of (startInclusive, endExclusive, step))
+  - Optional: The loss function (one of the LOSS_* constants)
+combo_scheme:
+  Seems to be similar to ranges. TODO: figure out how this works in more detail. Loss functions seem to never be specified.
```
Comment (dtracers) [inline bot_code/modelHelpers/actions/action_factory.py]: Combo takes a list of ranges and combines them into a single action using permutations.

Example Boost, Jump, Handbrake
All the permutations would be internally represented by a single number that the model will output.  
False, False, True = 1
False, False, False = 0

File: bot_code/modelHelpers/actions/dynamic_action_handler.py [hunk 2]
```diff
@@ -27,11 +31,13 @@ class DynamicActionHandler(SplitActionHandler):
     combo_name_index_map = {}
     action_sizes = []
     combo_action_sizes = []
+    actions = []  # Array of actions.
```
Comment (dtracers) [inline bot_code/modelHelpers/actions/dynamic_action_handler.py]: This is in the super class idk if we need to redefine it

File: bot_code/trainer/configs/randomised_trainer.cfg [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [Randomised Trainer Configuration]
 total_batches = 40000
 save_step = 50
-#teacher_package = TutorialBot.tutorial_bot_output
-teacher_package = TutorialBot.atba2_scorer
+#teacher_package =  models.TutorialBot.tutorial_bot_output
+teacher_package =  models.TutorialBot.atba2_scorer
```
Comment (dtracers) [inline bot_code/trainer/configs/randomised_trainer.cfg]: Did the tutorial bots gets moved inside models?
That might get confusing as they are not a model themselves.

File: bot_code/modelHelpers/actions/dynamic_action_handler.py (hunk 2 above)
Comment (DomNomNom) [reply bot_code/modelHelpers/actions/dynamic_action_handler.py]: Moved some thing to the super class

File: bot_code/trainer/configs/randomised_trainer.cfg (hunk 3 above)
Comment (DomNomNom) [reply bot_code/trainer/configs/randomised_trainer.cfg]: Removing this as we have a plan where to move TutorialBot instead. (group it close with FakeModel)

I'll squash these commits once you think we should submit it in this state.

**Pull request merged**